tests/unit_tests/regression/test_utilities.cpp

```diff
@@ -6,6 +6,19 @@
 
 namespace openturbine::tests {
 
+std::filesystem::path FindProjectRoot() {
+    std::filesystem::path currentPath = std::filesystem::current_path();
+
+    while (!currentPath.empty()) {
+        if (std::filesystem::exists(currentPath / "CMakeLists.txt")) {
+            return currentPath;
+        }
+        currentPath = currentPath.parent_path();
+    }
+
+    throw std::runtime_error("Could not find project root directory. CMakeLists.txt not found.");
+}
```

Comment:
Similar to my previous comment, this one will not work if the build directory is not a sub-directory of the source.

Reply:
Let's create an issue to harden this code later on.

Reply:
Agreed.